### PR TITLE
test: /analytics/get_quest_participation?id=${id}

### LINKED ISSUE #622: test: `/analytics/get_quest_participation?id=${id}`
read description here - https://github.com/starknet-id/starknet.quest/issues/602#issue-2255907921

### PR DESCRIPTION
**Pull request by EjembiEmmanuel**
# Pull Request type

Test

Please add the labels corresponding to the type of changes your PR introduces:

- Testing

Resolves: #622

### CODE REVIEW

File: tests/services/getQuestParticipant.test.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    });
+
+    const result = await getQuestsParticipation(undefined);
+    expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/analytics/get_quest_participation?id=undefined`
+    );
+    expect(result).toBeNull();
+  });
+
+  it("should handle no response", async () => {
```
Comment (ayushtom) [inline tests/services/getQuestParticipant.test.js]: this returns a valid response, can you handle this as a error 500 when the api fails to get data?

File: tests/services/getQuestParticipant.test.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    const result = await getQuestsParticipation(1);
+    expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/analytics/get_quest_participation?id=1`
+    );
+    expect(result).toEqual(mockData);
+  });
+
+  it("should handle fetch errors gracefully", async () => {
+    const mockResponse = "Quest not found";
```
Comment (ayushtom) [inline tests/services/getQuestParticipant.test.js]: not a valid response

Comment (ayushtom) [reply tests/services/getQuestParticipant.test.js]: you should be getting a empty array here ideally

File: tests/services/getQuestParticipant.test.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+global.fetch = jest.fn();
+
+describe("getQuestsParticipation", () => {
+  beforeEach(() => {
+    fetch.mockClear();
+  });
+
+  it("should fetch and return data for a valid id", async () => {
+    const mockData = {
```
Comment (ayushtom) [inline tests/services/getQuestParticipant.test.js]: i would suggest using the api once to see what is the ideal response you get from the api for this and also add that as a return type in the apiService.ts

**Comment by ayushtom** [comment]
hey i would also need you to add the test cases to apiServices.test.js

**Comment by ayushtom** [comment]
test cases look good to me

**Comment by EjembiEmmanuel** [comment]
@ayushtom I've added the test cases to apiServices.test.js

File: services/apiService.ts [hunk 4]
```diff
@@ -293,7 +293,9 @@ export const getQuestActivityData = async (id: number) => {
   }
 };
 
-export const getQuestsParticipation = async (id: number) => {
+export const getQuestsParticipation = async (
+  id: number
+): Promise<object[] | undefined> => {
```
Comment (ayushtom) [inline services/apiService.ts]: you need to add the type in backTypes.ts like - 
type QuestParticipants = {
  name: String;
  desc: String;
participants: String;
}[];

and import that type here as Promise<QuestParticipants>

**Comment by ayushtom** [comment]
Also you can remove the `tests/services/getQuestParticipant.test.js` from changelog

File: services/apiService.ts (hunk 4 above)
Comment (EjembiEmmanuel) [reply services/apiService.ts]: Alright

**Comment by EjembiEmmanuel** [comment]
@ayushtom I've effected the changes

**Comment by ayushtom** [comment]
@EjembiEmmanuel your build is failing

**Comment by EjembiEmmanuel** [comment]
@ayushtom please can you specify why it is failing

**Comment by EjembiEmmanuel** [comment]
@ayushtom I think I've fixed it. The issue was caused by the QuestParticipation type I declared in backTypes.d.ts. I had to refactor the usage in starknet.quest/app/analytics/[questId]/page.tsx to be compatible with it.

**Comment by ayushtom** [comment]
there was a build issue. I resolve it in my commit

**Comment by EjembiEmmanuel** [comment]
Okay

**Review by ayushtom — APPROVED** [review]
lgtm

**Pull request merged**